### DISCUSSION TITLE
Hoppscotch @ Github Externship Winter Cohort

### DISCUSSION THREAD

**Discussion by AndrewBastin** [hoppscotch/hoppscotch#1988] [Q&A]
# **GitHub Externship @ Hoppscotch**

**We ❤️ open source**

We are super excited to be a Partner Organization of the GitHub Externship Winter Cohort and look forward to supporting all interested students to participate and contribute to our [open source projects](https://github.com/hoppscotch).

This document covers the process, expectations and guidelines for any interested student so that we can guide, mentor in the best possible manner and most importantly review your community engagement which is a critical factor along with your project idea to be selected from our organization.

**This post may be updated from time to time. So please keep an eye on it. 👀**

Before you jump in and start contributing, here are a few documents from the GitHub Externship Team which we advise every student to read through carefully and thoroughly:

- [GitHub Externship: Website](https://externship.github.in/externship)
- [GitHub Externship: Timeline](https://externship.github.in/timeline)
- [Github Externship: Frequently Asked Questions](https://externship.github.in/faq)

Our awesome community members frequently request features on our [Discord server](https://hoppscotch.io/discord), [Telegram group](https://hoppscotch.io/telegram), [GitHub issues tab](https://github.com/hoppscotch/hoppscotch/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) and we have a never-ending list of awesome ideas. So please do check those places out for inspiration. Just don't forget about the 3-month span you will be having to work with to complete your goals.

## **Ideas**

Just to get your ideas up and running, we have assembled together some suggestions to help your search for proposals on what to work on for the Externship.

### **1. Existing issues:**

Have a look into the [Github issues page](https://github.com/hoppscotch/hoppscotch/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) of this repository and look out for existing issues. If you found issues that you think you can implement, then that's amazing! Feel free to pick them up (make sure no one is actively assigned to the issue though).

If we find some issues which are ideal for being worked on during the Externship period, we will mark it with the "neat for externship" label which you may want to look out for!

### **2. New features:**

Couldn't find any issue that piqued your interest? Well, the next option would be to make a new feature request (or any other kind of issue type) yourself! Try using Hoppscotch and if you find anything you felt like it was missing, then well open an issue as your submission to work on the project! (fun fact: that is how basically more than half of the features of Hoppscotch were built!)

### **3. Look into our other projects:**

Hoppscotch has many other projects in [our organization](https://github.com/hoppscotch). If browser extensions or a proxy server or writing and maintaining documentation is your deal, then please look into those as well!

## **Apply with a proposal**
1. Start here: [Externship Guidelines](https://github.com/hoppscotch/hoppscotch/discussions/1988)
2. Create a new discussion thread under [GitHub Externship Discussions](https://github.com/hoppscotch/hoppscotch/discussions/categories/externship) for your proposal - use [proposal template gist](https://gist.github.com/liyasthomas/2ece091639028a7ecaafaade5ae8e2c6) (copy & paste `RAW` template in new discussion)
3. Ask any doubts about submitting proposals under this thread

## **Built with**

- [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML)
- [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS), [SCSS](https://sass-lang.com), [Tailwind CSS](https://tailwindcss.com)
- [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
- [TypeScript](https://www.typescriptlang.org)
- [Vue](https://vuejs.org)
- [Nuxt](https://nuxtjs.org)

## **Contributing**

Please contribute using [GitHub Flow](https://guides.github.com/introduction/flow). Create a branch, add commits, and [open a pull request](https://github.com/hoppscotch/hoppscotch/compare).

Please read [`CONTRIBUTING`](CONTRIBUTING.md) for details on our [`CODE OF CONDUCT`](CODE_OF_CONDUCT.md), and the process for submitting pull requests to us.

## **Quick links**

[**`Repository →`**](https://github.com/hoppscotch/hoppscotch)

[**`Web app →`**](https://hoppscotch.io)

[**`Discord server →`**](https://hoppscotch.io/discord)

[**`Telegram group →`**](https://hoppscotch.io/externship_telegram)

## **Discussions channel**

We use [GitHub Discussions](https://github.com/hoppscotch/hoppscotch/discussions) as the primary channel for anything and everything related to GitHub Externship, ranging from

- Queries on issues you’re working or thinking about to take up.
- Mentor advice on your project ideas [Google Drive document once there is enough clarity for easier collaboration and feedback – check project proposal ideas above]
- Helping out your friends to onboard and contribute
- Finally even selection updates etc.

Before asking a question please check this [How do I ask a good question?](https://stackoverflow.com/help/how-to-ask)

## **Need help?**

Start a discussion [here](https://github.com/hoppscotch/hoppscotch/discussions) if you’re unable to get a response from the [Telegram](https://hoppscotch.io/externship_telegram), [Discord](https://hoppscotch.io/discord) communities.

Start a new idea under [discussions](https://github.com/hoppscotch/hoppscotch/discussions) only once you’ve finalized your project proposal.

Send us an email at support@hoppscotch.io for any administrative or operational concerns regarding the program.

## **Submit your proposal on time**

You can find the complete GitHub Externship program timeline [here](https://externship.github.in/timeline). Students can begin registering and submitting project proposals to partner organizations—including Hoppscotch.

All proposals must be submitted by December 14, 2021.

## **Mentors**

- [Andrew Bastin](https://github.com/andrewbastin)
- [Liyas Thomas](https://github.com/liyasthomas)

If you have any questions or queries please contact us via [Telegram](https://hoppscotch.io/telegram), [Discord](https://hoppscotch.io/discord) or send an email to us at support@hoppscotch.io.

We look forward to seeing your proposals!

Happy hacking! 🧑‍💻👩‍💻💖

**Comment by ashwinexe** [comment]
Hi, I have a couple of doubts:
1. I have fair knowledge with HTML, CSS and JS and I am really keen to learn Nuxt and TS. that also means I'm beginner in Nuxt Vue and TS. Does that still qualifies me to try for externship?

**Reply by liyasthomas** [reply]
Hi @ashwinexe, feel free to apply for the externship.

**Comment by RishabhAgarwal-2001** [comment]
Hey! quoting from [here](https://externship.github.in/organization/hoppscotch/61a241454d2ce4c5137e1448)

> Evaluation Criteria
> 
> Create a new discussion thread under GitHub Externship Discussions for your proposal - use proposal template gist (copy & paste RAW template in new discussion)
> 
> Complete the above task and submit it on our portal with proper implementation and screenshots in PDF Format.
> 

What does "completing the above task"  means here?

**Reply by liyasthomas** [reply]
Here (https://externship.github.in/organization/hoppscotch) you can find an option to submit your proposal as a PDF file upload. Simply copy your proposal content in a document and export it as PDF and upload it there.

**Reply by RishabhAgarwal-2001** [reply]
Got it!

**Comment by devblin** [comment]
Hello mentors @AndrewBastin @liyasthomas , I wanted to know if shortlisted candidates received any kind of mail before official announcement ?

**Reply by liyasthomas** [reply]
Selected candidates will receive acknowledgement from GitHub India or Reskill.
